### PR TITLE
Feat OIDC

### PR DESCRIPTION
**Pull request by sesha-d11**
<div id='description'>
This pull request significantly enhances OpenID Connect (OIDC) support by introducing new endpoints, services, and DAOs for client management, authorization flows, and token handling. It adds comprehensive validation, improves error handling, and includes extensive integration tests covering various scenarios including error conditions and edge cases.

</div>

### CODE REVIEW

File: src/test/resources/test.sql [hunk 1]
```diff
@@ -10,6 +10,8 @@ insert into token_config (tenant_id, algorithm, issuer, rsa_keys, access_token_e
 insert into user_config (tenant_id, host, port, is_ssl_enabled, get_user_path, create_user_path, authenticate_user_path, add_provider_path) values ('tenant1', 'localhost', 9090, 0, '/user', '/user', '/authenticateUser', '/provider') ON DUPLICATE KEY UPDATE tenant_id = "tenant1";
 insert into contact_verify_config (tenant_id, is_otp_mocked, whitelisted_inputs) values ('tenant1', 1, '{}') ON DUPLICATE KEY UPDATE tenant_id = "tenant1";
 insert into admin_config (tenant_id, username, password) values ('tenant1', 'admin', 'admin123') ON DUPLICATE KEY UPDATE tenant_id = "tenant1";
+insert into scope (tenant_id, name, display_name, description, icon_url, claims, is_oidc) values ('tenant1', 'profile', 'Profile', 'Profile scope for user information', "", json_array("name", "family_name", "given_name"), 1);
```
Comment (akshaypatidar1999) [inline src/test/resources/test.sql:13]: @BitoAgent Which version of mysql client I am using?

**Review by d11-animesh — APPROVED** [review]
